config/kernel-vfs-rw-iterate.m4

```diff
@@ -32,15 +32,16 @@ dnl #
 dnl # Linux 4.1 API
 dnl #
 AC_DEFUN([ZFS_AC_KERNEL_NEW_SYNC_READ],
-	[AC_MSG_CHECKING([whether new_sync_read() is available])
+	[AC_MSG_CHECKING([whether new_sync_read()/new_sync_write() are available])
 	ZFS_LINUX_TRY_COMPILE([
 		#include <linux/fs.h>
 	],[
-		new_sync_read(NULL, NULL, 0, NULL);
+		(void) new_sync_read(NULL, NULL, 0, NULL);
+		(void) new_sync_write(NULL, NULL, 0, NULL);
```

Comment:
If we're going to fix this we might as well go all the way to avoid future problems and pass typed values so this is caught at configure time.  How about this (untested) version.

```c
        ssize_t ret __attribute__ ((unused));
        struct file *filp = NULL;
        char __user *rbuf = NULL;
        const char __user *wbuf = NULL;
        size_t len = 0;
        loff_t ppos;

        ret = new_sync_read(filp, rbuf, len, &ppos);
        ret = new_sync_write(filp, wbuf, len,, &ppos);
```

Reply:
I like that better. The last line has `,,` in the parameter list, but that is why we have testing. ;)